core/tx_list.go

```diff
@@ -696,18 +696,19 @@ func (l *txPricedList) Underpriced(tx *types.Transaction) bool {
 }
 
 func (l *txPricedList) underpricedForMulti(h *multiCurrencyPriceHeap, tx *types.Transaction) bool {
-	underpriced := l.underpricedFor(h.nativeCurrencyHeap, tx)
+	// Has to be underpriced for ALL heaps to be underpriced for the whole pool.
+	underpriced := l.underpricedFor(h.nativeCurrencyHeap, tx, h.IsCheaper)
 	for _, sh := range h.currencyHeaps {
-		if l.underpricedFor(sh, tx) {
-			underpriced = true
+		if !l.underpricedFor(sh, tx, h.IsCheaper) {
+			underpriced = false
```

Comment:
With the above we could short-circuit here, right?

Reply:
yeap, changing it